pennylane/ops/qubit/hamiltonian.py

```diff
@@ -184,6 +185,14 @@ def __init__(
         method="rlf",
         id=None,
     ):
+        if qml.operation.active_new_opmath():
+            warn(
+                "Using qml.Hamiltonian with new operator arithmetic enabled is deprecated. To "
+                "suppress this warning, use 'qml.operation.disable_new_opmath()' to enable legacy "
+                "operator arithmetic before using qml.Hamiltonian.",
```

Comment:
```suggestion
                "operator arithmetic before using qml.ops.Hamiltonian.",
```

Reply:
If the user disables new op math they can just use `qml.Hamiltonian`

Reply:
The warning only be raised if someone tries to create a `Hamiltonian` with new op math enabled. If we're telling people to disable new op math then we don't need to raise the warning as there's no concern about mixing old and new op math.

Reply:
Yeah but if new opmath is enabled, `qml.Hamiltonian` will point to `LinearCombination` 🙃

Reply:
Yeah, but in the warning I'm saying that to use the legacy Hamiltonian without the warning you have to `qml.operation.disable_new_opmath()` before using `qml.Hamiltonian`

Reply:
I find this message somewhat confusing too because the reason the user is getting this message is precisely because they were using `qml.ops.Hamiltonian` instead of `qml.Hamiltonian`. If they were using `qml.Hamiltonian` with new op math enabled they would've gotten `LinearCombination`.

Reply:
I guess the first sentence of the warning message should probably be "Using qml.ops.Hamiltonian with new operator arithmetic enabled is deprecated."

Reply:
How about something like:

```
"Using the Hamiltonian class with new operator arithmetic is deprecated. 
Some functionality may not work as expected, and use of the Hamiltonian class with 
the new operator arithmetic will not be supported in future releases of PennyLane.

You can update your code to the new operator arithmetic by using qml.Hamiltonian 
instead of importing the Hamiltonian class directly. When the new operator arithmetic is enabled, 
`qml.Hamiltonian` will access the new corresponding implementation. 

Alternatively, to continue accessing the legacy functionality, you can use 
'qml.operation.disable_new_opmath()'."
```

Reply:
> How about something like:

Could the user have used `qml.ops.Hamiltonian` because they didn't know that it would be different from `qml.Hamiltonian`, as opposed to intentionally looking for the old `Hamiltonian`? Should we explain this as well? It's not so common that when you import the same thing from different places you get completely different things.